OnlineShop.Services.Catalog.Infrastructure/Repositories/Implementations/ProductRepository.cs

```diff
@@ -0,0 +1,80 @@
+﻿using MongoDB.Driver;
+using OnlineShop.Services.Catalog.Application.Exceptions;
+using OnlineShop.Services.Catalog.Domain.Models.Data;
+using OnlineShop.Services.Catalog.Domain.Repositories.Interfaces;
+using OnlineShop.Services.Catalog.Infrastructure.Data.Interfaces;
+using System.Linq.Expressions;
+
+namespace OnlineShop.Services.Catalog.Infrastructure.Repositories.Implementations
+{
+    public class ProductRepository : IProductRepository
+    {
+        private readonly ICatalogContext _catalogContext;
+
+        public ProductRepository(ICatalogContext catalogContext)
+        {
+            _catalogContext = catalogContext;
+        }
+
+        public async Task<IEnumerable<Product>> GetAllAsync()
+            => await _catalogContext
+                     .Products
+                     .Find(p => true)
+                     .ToListAsync();
+
+        public async Task<Product> GetAsync(string id)
+        {
+            try
+            {
+                return await _catalogContext
+                            .Products
+                            .Find(product => product.Id == id)
+                            .FirstOrDefaultAsync();
+            }
+            catch(FormatException ex)
+            {
+                throw new InvalidIdException(ex.Message);
+            }
```

Comment:
how to do it in a right way then? Taking into account, that MongoDB throws FormatException if you're trying to pass incorrect format of hex string

Reply:
Since you have middleware handling errors, there's no need to use try-catch blocks in repositories or services. You can perform all necessary checks in validators or services